packages/backend-common/src/scm/git.test.ts

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+jest.mock('isomorphic-git');
+jest.mock('isomorphic-git/http/node');
+jest.mock('fs-extra');
+
+import * as isomorphic from 'isomorphic-git';
+import * as Git from './git';
+import http from 'isomorphic-git/http/node';
+import fs from 'fs-extra';
+
+describe('Git', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+  describe('add', () => {
+    it('should call isomorphic-git add with the correct arguments', async () => {
+      const git = Git.fromAuth({});
+      const dir = 'mockdirectory';
+      const filepath = 'mockfile/path';
+
+      await git.add({ dir, filepath });
```

Comment:
Looks like pretty much all of these ops are rooted in a dir, makes me wonder if the interface could be made differently somehow. Like `Git.init({ path, etcetc }).add({...}).commit({...})...`. Where there is a small set of initial static functions like `init` and `open` or whatnot and the return value is like a `Repository` or similar.

Reply:
Yeah was thinking about that as well, although it would be a bit more wrapping going on. Not really needed atm though tbh, and a refactor we could do in the future if we think it's needed